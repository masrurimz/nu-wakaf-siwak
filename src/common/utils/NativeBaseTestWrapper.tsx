import { NativeBaseProvider } from 'native-base';
import React from 'react';

interface NativeBaseTesstWrapperProps {
  children: React.ReactNode;
}

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const NativeBaseTestWrapper = (props: NativeBaseTesstWrapperProps) => {
  const { children } = props;

  return (
    <NativeBaseProvider initialWindowMetrics={inset}>
      {children}
    </NativeBaseProvider>
  );
};
