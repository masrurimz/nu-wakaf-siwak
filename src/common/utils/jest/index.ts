export * from './renderWithUtils';
export * from './QueryClientProviderTestWrapper';
export * from './NativeBaseProviderTestWrapper';
