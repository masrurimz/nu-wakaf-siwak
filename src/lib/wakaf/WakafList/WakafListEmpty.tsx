import { Heading, Text, VStack } from 'native-base';
import React from 'react';
import EmptyListIllustration from 'src/app/assets/illustrations/EmptyListIllustration.svg';

interface WakafListEmptyProps {
  searchQuery: string;
}

const WakafListEmpty = (props: WakafListEmptyProps) => {
  const { searchQuery } = props;

  const heading = 'Tidak ada data';
  const description = searchQuery
    ? `Pencarian dengan "${searchQuery}" tidak menemukan apapun`
    : 'Tap tombol tambahkan aset untuk menambahkan data';

  return (
    <VStack
      flex={0.7}
      alignItems="center"
      justifyContent="center"
      space={5}
      px={5}>
      <EmptyListIllustration height={200} width={200} />
      <VStack space={2}>
        <Heading textAlign="center">{heading}</Heading>
        <Text textAlign="center">{description}</Text>
      </VStack>
    </VStack>
  );
};

export default WakafListEmpty;
