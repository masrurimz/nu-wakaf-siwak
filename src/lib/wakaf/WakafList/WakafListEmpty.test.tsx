import { screen } from '@testing-library/react-native';
import React from 'react';
import { renderWithUtils } from '../../../common/utils';
import WakafListEmpty from './WakafListEmpty';

describe('Unit Test WakafListEmpty.tsx ', () => {
  it('should render component correcty', () => {
    renderWithUtils(<WakafListEmpty searchQuery="" />);

    const tree = screen.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should display 'Tap tombol tambahkan aset untuk menambahkan data' when empty search query given", () => {
    renderWithUtils(<WakafListEmpty searchQuery="" />);

    expect(
      screen.getByText('Tap tombol tambahkan aset untuk menambahkan data'),
    ).toBeTruthy();
  });

  it("should display 'Pencarian dengan kata kunci 'masjid' tidak menemukan apapun' when 'masjid' search query given", () => {
    renderWithUtils(<WakafListEmpty searchQuery="masjid" />);

    expect(
      screen.getByText(
        'Pencarian dengan kata kunci "masjid" tidak menemukan apapun',
      ),
    ).toBeTruthy();
  });
});
