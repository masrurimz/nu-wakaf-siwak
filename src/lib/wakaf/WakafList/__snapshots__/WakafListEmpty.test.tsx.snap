// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test WakafListEmpty.tsx  should render component correcty 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "flex": 0.7,
        "justifyContent": "center",
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <SvgMock
      height={200}
      width={200}
    />
    <View
      dataSet={Object {}}
      style={
        Object {
          "height": 20,
        }
      }
    />
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#27272a",
            "fontFamily": "Inter-Bold",
            "fontSize": 24,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 33,
            "textAlign": "center",
            "textDecorationLine": undefined,
          }
        }
      >
        Tidak ada data
      </Text>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#27272a",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 19.25,
            "textAlign": "center",
            "textDecorationLine": undefined,
          }
        }
      >
        Tap tombol tambahkan aset untuk menambahkan data
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
